examples/complete/main.tf

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# Resource Group
+# (if var.resource_group is null, create a new RG using var.prefix)
+##############################################################################
+
+resource "ibm_resource_group" "resource_group" {
+  count = var.resource_group != null ? 0 : 1
+  name  = "${var.prefix}-rg"
+
+}
+
+data "ibm_resource_group" "existing_resource_group" {
+  count = var.resource_group != null ? 1 : 0
+  name  = var.resource_group
+}
+
+#############################################################################
+# Security Group
+#############################################################################
+
+data "ibm_is_security_group" "existing_sg" {
+  count = var.create_security_group ? 0 : 1
+  name  = var.security_group_id
+}
+
+resource "ibm_is_security_group" "sg" {
```

Comment:
the logic to optionally create the security group could be moved up in the root of the module rather than being just in this example.